### PR TITLE
#8 - support customization of replication values.

### LINKED ISSUE #8: Not possible to interpolate variables on keyspace replication
Hi everyone,

I'm trying to run the following script:
```
createKeyspace {
  version 1
  name keyspace
  author 'admin'
  description 'create keyspace test'
  replication {
    replication_factor replicationFactor
  }
}
```
```
def cassalog = new CassalogImpl(session: session)
    def replicationFactor = 3
    cassalog.execute(script, [keyspace: keyspace, replicationFactor: replicationFactor])
```

but it does not work. It runs smoothly but replaces my provided replicationFactor with the default value 1. Same happens with the replicationStrategy.

### PR DESCRIPTION
**Pull request by nachogiljaldo**
This PR should fix https://github.com/hawkular/cassalog/issues/8

### CODE REVIEW

File: src/main/groovy/org/cassalog/core/CreateKeyspace.groovy [hunk 1]
```diff
@@ -72,7 +72,7 @@ CREATE KEYSPACE $name WITH replication = { 'class': '${replication.strategy}', '
 
   void replication(Closure closure) {
     closure.delegate = replication
-    closure.resolveStrategy = Closure.DELEGATE_ONLY
+    closure.resolveStrategy = Closure.OWNER_FIRST
```
Comment (jsanda) [inline src/main/groovy/org/cassalog/core/CreateKeyspace.groovy:75]: Can you explain why the change to OWNER_FIRST? I am just curious because I am not writing a whole lot of Groovy these days, particularly meta programming :)

File: src/test/groovy/org/cassalog/core/CassalogTest.groovy [hunk 2]
```diff
@@ -104,6 +105,28 @@ class CassalogTest extends CassalogBaseTest {
     assertChangeSetEquals(rows[1], new ChangeSet(version: change2Id, author: 'admin', description: 'test'))
   }
 
+  @Test
```
Comment (jsanda) [inline src/test/groovy/org/cassalog/core/CassalogTest.groovy:108]: Thanks for the test. Looks good!

File: src/main/groovy/org/cassalog/core/CreateKeyspace.groovy (hunk 1 above)
Comment (nachogiljaldo) [reply src/main/groovy/org/cassalog/core/CreateKeyspace.groovy:75]: so, I'm not an expert either, but since the strategy is resolve from DELEGATE_ONLY and the delegate is replication, that results on assigning to replication the exact same value that replication has, which in the end resulted on my overriden properties not being honoured

File: src/test/groovy/org/cassalog/core/CassalogTest.groovy (hunk 2 above)
Comment (nachogiljaldo) [reply src/test/groovy/org/cassalog/core/CassalogTest.groovy:108]: I noticed the tests leave a lot of keyspaces in cassandra, should I change them so it leaves cassandra in the same exact status it was?

Comment (jsanda) [reply src/test/groovy/org/cassalog/core/CassalogTest.groovy:108]: There is the ```resetSchema``` method in CassalogBaseTest.groovy that is called in each test. Do you have something else in mind?

Comment (nachogiljaldo) [reply src/test/groovy/org/cassalog/core/CassalogTest.groovy:108]: Just added one more commit to ensure that the keyspaces created on the test are deleted after the test. Let me know if you do not like it and I can remove it.

**Comment by jsanda** [comment]
Thanks for the PR!

**Comment by nachogiljaldo** [comment]
hi @jsanda could you release the current version to repositories? Thanks in advance!

**Comment by jsanda** [comment]
Done!
